### ISSUE TITLE
WLED (ESP-8266)

### ISSUE THREAD

**Issue by ikrowni** [antonpup/Aurora#2027]
Please add support for the following device:

**Device Name:** WLED (ESP-8266)

**Device Website:** https://github.com/Aircoookie/WLED/wiki

This is heavily integrated with my current office setup through home assistant. I would love the ability to control the lights with my games. 

https://www.youtube.com/watch?v=es9rawSGUPk

**Comment by Cheerpipe** [comment]
I have some Chinese led strips that talks ledenet integrated with Aurora. Did you know what protocol talks your LEDs?

**Comment by ikrowni** [comment]
udp

Blynk

Mqtt

Hue

E1.31

Are all options available within the sync options.

Not sure if any of that helps

**Comment by ikrowni** [comment]
> Did you know what protocol talks your LEDs?

Does this page help?

https://github.com/Aircoookie/WLED/wiki/Add-own-functionality

**Comment by Operation760** [comment]
I second this for a formal enhancement and integration support for WLED. Support for WLED alone opens up some very cool, highly complimentary features/functions and is worth it alone. However, an integrated Aurora-WLED could extend Aurora into a larger home automation ecosystem and open up a lot of very cool possibilities for event based actions and synchronized lighting via applications and platforms like Home Assistant, ESPHome, Tasmota, etc. Possibilities for devices and services far beyond PC peripherals and the Razer-Hue-Aurora integration. 

Think Aurora changing PC lighting behavior to notify you that  someone or something has  triggered motion at your front door or the wife, via presence detection has not tripped a defined geozone exit alert and therefore game time has been extended ;). PC getting too hot? Game room temp increasing? Turn on the AC! 

Tons of possibilities! Also Aurora’s integrations could extend into the WLED and HA ecosystems. Being able to extend PC-basest conditions and events is a sought after use-case and within Aurora’s excellent existing capabilities.

**Comment by ikrowni** [comment]
The developer of the wled project just reach out and said that the JSON API would probably be the best option for an integration into Aurora.

https://github.com/Aircoookie/WLED/wiki/JSON-API

**Comment by nzbr** [comment]
I have created this script a while back for using my RGB led strips with Aurora. It was not created with WLED in mind, so it only uses the UDP interface but it works great for me.

To use it, just put the script in `C:\Program Files\Aurora\Scripts\Devices` as `UDP.cs`. Then, when you restart Aurora, a new device-type UDP should appear, where you can input the IP Adresses.

https://gist.github.com/nzbr/a88943aa1904decd298e5765c7f5aaa5

**Comment by ikrowni** [comment]
This looks awesome and I'll be trying it in the next 30m to see if it works for me. Is there a benefit to using the e1.31 over udp or any drawbacks to using udp as opposed to other methods? I honestly have no idea how any of it works so was just curious.

**Comment by ikrowni** [comment]
> I have created this script a while back for using my RGB led strips with Aurora. It was not created with WLED in mind, so it only uses the UDP interface but it works great for me.


After attempting to link the 2 services together using your script it doesnt seem to be working. After adding the script intot he folder you said i can see udp as an option and have went through the setup process. The only part im not really sure about is the option for (Key to Use) What is this and what should i have it set to?

After this setup is finished was there anything else that i need to do to get it working? Do i need to add anything to the profiles im using? Im pretty new to aurora so im still trying to figure it out. Thanks a million for your help though.

**Comment by nzbr** [comment]
Key to Use is the Key (in Aurora) of whose color will be sent to wled. It is set to Peripheral by default, so the led strip will habe the same color as your Mouse. The script does not use e1.31 over UDP but raw rgb bytes over UDP. WLED can receive rawudp on port 19446, which should be the default in the script.
When everything is set up correctly, it should work without modifying the profiles (it just uses the color of an existing key after all) , just enable & start the UDP device. If it works the WLED web-interface will display something like "WLED is receiving data from Hyperion"

**Comment by ikrowni** [comment]
Thanks this worked perfectly. Only downside is the 490 led limit

**Comment by nzbr** [comment]
Nice. I have not encountered that limit myself (The most LEDs I have on one strip is 300). It would still be nice to see UDP (and E1.31) device integrations built in to Aurora

**Comment by ikrowni** [comment]
Agreed. The limit is that udp only allows 490 from what u/aircookie over at the wled project said. Hopefully the Aurora devs take notice of this and get built in implementation of e1.31 so that we can have really large runs of LED lights going with Aurora. Thanks again for the script.

**Comment by Operation760** [comment]
Working perfectly here too once I expanded the settings window in the device manager. I couldn't see the IP address input field. Doh.

**Comment by TheHolyLoli** [comment]
> I have created this script a while back for using my RGB led strips with Aurora. It was not created with WLED in mind, so it only uses the UDP interface but it works great for me.
> 
> To use it, just put the script in `C:\Program Files\Aurora\Scripts\Devices` as `UDP.cs`. Then, when you restart Aurora, a new device-type UDP should appear, where you can input the IP Adresses.
> 
> https://gist.github.com/nzbr/a88943aa1904decd298e5765c7f5aaa5

does this only sends 1 color to all LEDS?

**Comment by ikrowni** [comment]
It is linked to a button of your choice. So if the button of your choice flips through multiple colors it will change through those colors.
